Mathlib/CategoryTheory/Products/Basic.lean

```diff
@@ -286,77 +288,86 @@ end Functor
 namespace NatTrans
 
 /-- The cartesian product of two natural transformations. -/
-@[simps]
-def prod {F G : A ⥤ B} {H I : C ⥤ D} (α : F ⟶ G) (β : H ⟶ I) : F.Prod H ⟶ G.Prod I
+@[simps!]
+def prod {F G : A ⥤ B} {H I : C ⥤ D} (α : F ⟶ G) (β : H ⟶ I) : F.prod H ⟶ G.prod I
     where
   app X := (α.app X.1, β.app X.2)
-  naturality' X Y f := by
+  naturality {X} {Y} f := by
     cases X; cases Y
-    simp only [functor.prod_map, Prod.mk.inj_iff, prod_comp]
-    constructor <;> rw [naturality]
+    simp only [Functor.prod_map, prod_comp] 
+    rw [Prod.mk.inj_iff] 
+    constructor 
+    repeat {rw [naturality]}
 #align category_theory.nat_trans.prod CategoryTheory.NatTrans.prod
 
 /- Again, it is inadvisable in Lean 3 to setup a notation `α × β`;
-   use instead `α.prod β` or `nat_trans.prod α β`. -/
+   use instead `α.prod β` or `NatTrans.prod α β`. -/
 end NatTrans
 
 /-- `F.flip` composed with evaluation is the same as evaluating `F`. -/
-@[simps]
+@[simps!]
 def flipCompEvaluation (F : A ⥤ B ⥤ C) (a) : F.flip ⋙ (evaluation _ _).obj a ≅ F.obj a :=
-  (NatIso.ofComponents fun b => eqToIso rfl) <| by tidy
+  (NatIso.ofComponents fun b => eqToIso rfl) <| by aesop_cat
 #align category_theory.flip_comp_evaluation CategoryTheory.flipCompEvaluation
 
 variable (A B C)
 
-/-- The forward direction for `functor_prod_functor_equiv` -/
-@[simps]
+/-- The forward direction for `functorProdFunctorEquiv` -/
+@[simps!]
 def prodFunctorToFunctorProd : (A ⥤ B) × (A ⥤ C) ⥤ A ⥤ B × C
     where
   obj F := F.1.prod' F.2
-  map F G f := { app := fun X => (f.1.app X, f.2.app X) }
+  map f := { app := fun X => (f.1.app X, f.2.app X) }
 #align category_theory.prod_functor_to_functor_prod CategoryTheory.prodFunctorToFunctorProd
 
-/-- The backward direction for `functor_prod_functor_equiv` -/
-@[simps]
+/-- The backward direction for `functorProdFunctorEquiv` -/
+@[simps!]
 def functorProdToProdFunctor : (A ⥤ B × C) ⥤ (A ⥤ B) × (A ⥤ C)
     where
-  obj F := ⟨F ⋙ CategoryTheory.prod.fst B C, F ⋙ CategoryTheory.prod.snd B C⟩
-  map F G α :=
+  obj F := ⟨F ⋙ CategoryTheory.Prod.fst B C, F ⋙ CategoryTheory.Prod.snd B C⟩
+  map α :=
     ⟨{  app := fun X => (α.app X).1
-        naturality' := fun X Y f => by
-          simp only [functor.comp_map, prod.fst_map, ← prod_comp_fst, α.naturality] },
+        naturality := fun X Y f => by
+          simp only [Functor.comp_map, Prod.fst_map, ← prod_comp_fst, α.naturality] },
       { app := fun X => (α.app X).2
-        naturality' := fun X Y f => by
-          simp only [functor.comp_map, prod.snd_map, ← prod_comp_snd, α.naturality] }⟩
+        naturality := fun X Y f => by
+          simp only [Functor.comp_map, Prod.snd_map, ← prod_comp_snd, α.naturality] }⟩
 #align category_theory.functor_prod_to_prod_functor CategoryTheory.functorProdToProdFunctor
 
-/-- The unit isomorphism for `functor_prod_functor_equiv` -/
-@[simps]
+/-- The unit isomorphism for `functorProdFunctorEquiv` -/
+-- @[simps!]
```

Comment:
Please leave a porting note that briefly motivates this change. If the `simps` can't be made to work, please manually add the simp lemmas that it would generated.

Reply:
Sorry. That was left over from experimentation. I also restored all possible declarations to `simps` from `simps!`